public_html/wp-content/plugins/wordcamp-payments/includes/sponsor-invoice.php

```diff
@@ -419,8 +419,85 @@ function save_invoice( $post_id, $post ) {
 			update_post_meta( $post_id, $meta_key, $value );
 		}
 	}
+
+	$should_schedule_invoice_reminder = apply_filters( 'should_schedule_invoice_reminder', true );
+	if ( $should_schedule_invoice_reminder && 'wcbsi_approved' === $post->post_status ) {
+		// Schedule organizer reminder if its not already scheduled.
+		$args = array( $post_id );
+		if ( ! wp_next_scheduled( 'send_invoice_pending_reminder', $args ) && ! get_post_meta( $post_id, 'last_reminder_details', true ) ) {
+			wp_schedule_single_event( time() + DAY_IN_SECONDS * 30, 'send_invoice_pending_reminder', array( $post_id ) );
+			update_post_meta( $post_id, 'reminder_schedule_start_time', time() );
+		}
+	}
```

Comment:
I have switched to a single cron job that you and @coreymckrill suggested. That does makes more sense, and since we have a index table its also good in terms of performance. Resolving this discussion now.

Reply:
Also for older post, I am assuming the current time to be the time reminder sent at, because I do not think we do have any better information that we can use to assume.